### PR TITLE
Content Edit: Implement Monaco Editor

### LINKED ISSUE #25946: Content Edit: Implement Monaco Editor
### Parent Issue

https://github.com/dotCMS/core/issues/25445

### Task

We need to develop a component that allows us to open a dialog with the Monaco Editor.

### Proposed Objective

Core Features

### Proposed Priority

Priority 3 - Average

### Acceptance Criteria

- Develop a clickable button that opens a dialog with the Monaco Editor once is clicked

### External Links... Slack Conversations, Support Tickets, Figma Designs, etc.

_No response_

### Assumptions & Initiation Needs

Use  the [ngx monaco editor lib](https://www.npmjs.com/package/ngx-monaco-editor) and PrimeNg

### Quality Assurance Notes & Workarounds

_No response_

### Sub-Tasks & Estimates

_No response_

### PR DESCRIPTION
**Pull request by rjvelazco**
### Proposed Changes
* Implement Monaco Editor

### Checklist
- [x] Tests

### <samp>🤖 Generated by Copilot at 0fc6bc6</samp>

Added a code editor feature to the `binary-field` component. This feature allows the user to write custom code for the binary field in a dialog. The changes include the component template, style, test, story and ts files.

## Related Issue
Fixes #25946 

# Explanation of Changes

### <samp>🤖 Generated by Copilot at 0fc6bc6</samp>

*  Add a feature to the binary field component that allows the user to write custom code for the field in a dialog ([link](https://github.com/dotCMS/core/pull/25955/files?diff=unified&w=0#diff-fe43ae6cefd137f07a1883a5f279da1a1775ea58561fcc1f770a3a617d65d40eR1-R5), [link](https://github.com/dotCMS/core/pull/25955/files?diff=unified&w=0#diff-740e087d4ab3dc50c530488107c47d9fd0da08096e92f5459e56698387a23554R1-R30))
* Add a test file for the binary field component to verify its creation ([link](https://github.com/dotCMS/core/pull/25955/files?diff=unified&w=0#diff-492b94bd17fb323e956e8588fb4f73406ce1000e53477e305ab1888247eb8e8aR1-R22))
* Add a story file for the binary field component to display its default state ([link](https://github.com/dotCMS/core/pull/25955/files?diff=unified&w=0#diff-a939da2218f476e5f8f219886765fa9fc0ceeaa94fe280e8a75a04d9c3886768R1-R34))

### Screenshot

https://github.com/dotCMS/core/assets/72418962/1398ea6f-69f2-488f-8d10-6e5e590e5f41

### CODE REVIEW

File: core-web/libs/contenttype-fields/tsconfig.json [hunk 1]
```diff
@@ -1,14 +1,5 @@
 {
-    "compilerOptions": {
```
Comment (hmoreras) [inline core-web/libs/contenttype-fields/tsconfig.json:2]: why delete these, is part of the `tsconfig.base.json` now ?

**Review by hmoreras — APPROVED** [review]

File: core-web/libs/contenttype-fields/src/lib/fields/binary-field/binary-field.component.html [hunk 2]
```diff
@@ -0,0 +1,10 @@
+<p-button class="binary-field__code-button" (click)="showCodeEditor()" data-testId="code-button">{{
```
Comment (zJaaal) [inline core-web/libs/contenttype-fields/src/lib/fields/binary-field/binary-field.component.html:1]: does the class is being used?

File: core-web/libs/contenttype-fields/src/lib/fields/binary-field/binary-field.component.ts [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+    changeDetection: ChangeDetectionStrategy.OnPush
+})
+export class BinaryFieldComponent implements OnInit {
+    public labelCode!: string;
+    visible = false;
+
+    constructor(private dotMesagePipe: DotMessagePipe) {}
+
+    ngOnInit() {
+        this.labelCode = this.dotMesagePipe.transform('contenttypes.content.edit.write.code');
```
Comment (zJaaal) [inline core-web/libs/contenttype-fields/src/lib/fields/binary-field/binary-field.component.ts:25]: why don't we use the pipe directly or `DotMessageService`?

**Review by zJaaal — CHANGES_REQUESTED** [review]

File: core-web/libs/contenttype-fields/src/lib/fields/binary-field/binary-field.component.ts (hunk 3 above)
Comment (rjvelazco) [reply core-web/libs/contenttype-fields/src/lib/fields/binary-field/binary-field.component.ts:25]: The label is being used in the button and in the header. That's why I created a variable.

File: core-web/libs/contenttype-fields/src/lib/fields/binary-field/binary-field.component.html (hunk 2 above)
Comment (rjvelazco) [reply core-web/libs/contenttype-fields/src/lib/fields/binary-field/binary-field.component.html:1]: No, I will delate it

File: core-web/libs/contenttype-fields/tsconfig.json (hunk 1 above)
Comment (rjvelazco) [reply core-web/libs/contenttype-fields/tsconfig.json:2]: Yes! Now it's part of the `tsconfig.base.json`.

**Review by zJaaal — APPROVED** [review]

File: core-web/libs/contenttype-fields/src/lib/fields/binary-field/binary-field.component.ts [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+    constructor(private readonly dotMessageService: DotMessageService) {}
+
+    ngOnInit() {
+        this.labelCode = this.dotMessageService.get('contenttypes.content.edit.write.code');
+    }
+
+    showCodeEditor() {
+        this.visible = true;
+    }
+}
```
Comment (fmontes) [inline core-web/libs/contenttype-fields/src/lib/fields/binary-field/binary-field.component.ts:31]:
```suggestion
export class BinaryFieldComponent {
    dotMessageService = inject(DotMessageService);
    labelCode = this.dotMessageService.get('contenttypes.content.edit.write.code');
    visible = false;

    showCodeEditor() {
        this.visible = true;
    }
}
```

Comment (fmontes) [reply core-web/libs/contenttype-fields/src/lib/fields/binary-field/binary-field.component.ts:31]: New angular feature simpiply stuff :)

File: core-web/libs/contenttype-fields/src/lib/fields/binary-field/binary-field.component.spec.ts [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+                useValue: CONTENTTYPE_FIELDS_MESSAGE_MOCK
+            }
+        ]
+    });
+
+    beforeEach(() => (spectator = createComponent()));
+
+    it('should create', () => {
+        expect(spectator.component).toBeTruthy();
+    });
```
Comment (fmontes) [inline core-web/libs/contenttype-fields/src/lib/fields/binary-field/binary-field.component.spec.ts:41]: Uneeded

**Review by fmontes — CHANGES_REQUESTED** [review]

File: core-web/libs/contenttype-fields/src/lib/fields/binary-field/binary-field.component.ts (hunk 4 above)
Comment (oidacra) [reply core-web/libs/contenttype-fields/src/lib/fields/binary-field/binary-field.component.ts:31]: You can do too `labelCode = inject(DotMessageService).get('contenttypes.content.edit.write.code');`

**Review by zJaaal — APPROVED** [review]

**Review by oidacra — APPROVED** [review]

File: core-web/libs/contenttype-fields/src/lib/fields/binary-field/binary-field.component.ts (hunk 4 above)
Comment (rjvelazco) [reply core-web/libs/contenttype-fields/src/lib/fields/binary-field/binary-field.component.ts:31]: @fmontes Really cool! Thanks

File: core-web/libs/contenttype-fields/src/lib/fields/binary-field/binary-field.component.spec.ts (hunk 5 above)
Comment (rjvelazco) [reply core-web/libs/contenttype-fields/src/lib/fields/binary-field/binary-field.component.spec.ts:41]: Sure

File: core-web/libs/contenttype-fields/src/lib/fields/binary-field/binary-field.component.ts (hunk 4 above)
Comment (rjvelazco) [reply core-web/libs/contenttype-fields/src/lib/fields/binary-field/binary-field.component.ts]: Done!

**Review by fmontes — APPROVED** [review]

**Pull request merged**